src/app/submission/form/collection/submission-form-collection.component.ts

```diff
@@ -0,0 +1,193 @@
+import { JsonPatchOperationsBuilder } from '../../../core/json-patch/builder/json-patch-operations-builder';
+import { PaginatedList } from '../../../core/data/paginated-list';
+import { SubmissionService } from '../../submission.service';
+import { SubmissionObject } from '../../../core/submission/models/submission-object.model';
+import { SubmissionJsonPatchOperationsService } from '../../../core/submission/submission-json-patch-operations.service';
+
+interface CollectionListEntryItem {
+  id: string;
+  name: string;
+}
+
+interface CollectionListEntry {
+  communities: CollectionListEntryItem[],
+  collection: CollectionListEntryItem
+}
+
+@Component({
+  selector: 'ds-submission-form-collection',
+  styleUrls: ['./submission-form-collection.component.scss'],
+  templateUrl: './submission-form-collection.component.html'
+})
+export class SubmissionFormCollectionComponent implements OnChanges, OnInit {
+  @Input() currentCollectionId: string;
+  @Input() currentDefinition: string;
+  @Input() submissionId;
+
+  /**
+   * An event fired when a different collection is selected.
+   * Event's payload equals to new SubmissionObject.
+   */
+  @Output() collectionChange: EventEmitter<SubmissionObject> = new EventEmitter<SubmissionObject>();
+
+  public disabled$ = new BehaviorSubject<boolean>(true);
+  public model: any;
+  public searchField: FormControl = new FormControl();
+  public searchListCollection$: Observable<CollectionListEntry[]>;
+  public selectedCollectionId: string;
+  public selectedCollectionName$: Observable<string>;
+
+  protected pathCombiner: JsonPatchOperationPathCombiner;
+  private scrollableBottom = false;
+  private scrollableTop = false;
+  private subs: Subscription[] = [];
+
+  constructor(protected cdr: ChangeDetectorRef,
+              private communityDataService: CommunityDataService,
+              private operationsBuilder: JsonPatchOperationsBuilder,
+              private operationsService: SubmissionJsonPatchOperationsService,
+              private submissionService: SubmissionService) {
+  }
+
+  @HostListener('mousewheel', ['$event']) onMousewheel(event) {
+    if (event.wheelDelta > 0 && this.scrollableTop) {
+      event.preventDefault();
+    }
+    if (event.wheelDelta < 0 && this.scrollableBottom) {
+      event.preventDefault();
+    }
+  }
+
+  onScroll(event) {
+    this.scrollableBottom = (event.target.scrollTop + event.target.clientHeight === event.target.scrollHeight);
+    this.scrollableTop = (event.target.scrollTop === 0);
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (hasValue(changes.currentCollectionId)
+      && hasValue(changes.currentCollectionId.currentValue)) {
+      this.selectedCollectionId = this.currentCollectionId;
+
+      // @TODO replace with search/top browse endpoint
+      // @TODO implement community/subcommunity hierarchy
+      const communities$ = this.communityDataService.findAll().pipe(
```

Comment:
Am I right in the assumption that this would only show the 20 (the first page) of communities, and for each community the first 20 collections. So, that as it is now, you can't submit to a collection that isn't on that first page?

I ask because @LotteHofstede  faced the same problem when designing the context sensitive admin menu popups, and as far as I can tell, it can't be done yet with the rest api as it is now (short of fetching _all_ communities and collections and searching the tree on the client)

Reply:
you are right, I was deceived by FindAll method. If you are agree we could resolve this with a separate issue, maybe when we will have a rest endpoint for this feature

Reply:
I agree to solve this separately, that's why I suggested that search-as-you-type rest endpoint in the last meeting

For now, that collection search can still work if you simply query the discovery endpoint directly and filter by dso type. [Here](https://github.com/DSpace/dspace-angular/pull/367/files#diff-eacbb9b4341c4775a53c58ad82a90870R84) is an example from @LotteHofstede's PR. This has two downsides: 
- You need to type whole words before the discovery endpoint returns results. 
- You also can't show the parent communities, because those come with their child collections embedded, so we end up downloading way too much to simply show those parents. I'd reconsider that, if we have support for partial objects.

So the usability leaves some things to be desired, but it will at least work with all collections.

Reply:
created an [issue](https://github.com/DSpace/dspace-angular/issues/373) for this